mvc/src/main/java/com/interface21/webmvc/servlet/mvc/tobe/AnnotationHandlerMapping.java

```diff
@@ -20,10 +30,58 @@ public AnnotationHandlerMapping(final Object... basePackage) {
     }
 
     public void initialize() {
+        Reflections reflections = new Reflections(basePackage);
+        Set<Class<?>> controllerClasses = reflections.getTypesAnnotatedWith(Controller.class);
+        controllerClasses.forEach(this::register);
         log.info("Initialized AnnotationHandlerMapping!");
     }
 
     public Object getHandler(final HttpServletRequest request) {
-        return null;
+        String requestURI = request.getRequestURI();
+        String requestMethod = request.getMethod();
+        HandlerKey handlerKey = new HandlerKey(requestURI, RequestMethod.of(requestMethod));
+        HandlerExecution handlerExecution = handlerExecutions.get(handlerKey);
+        if (Objects.isNull(handlerExecution)) {
```

Comment:
`handlerExecution` 값을 이미 `get`으로 뽑아왔다면 `Objects.isNull`을 쓰지 않아도 되지 않을까요?

```suggestion
        if (handlerExecution == null) {
```

위와 같은 비교를 통해 불필요한 유틸리티 사용을 피할 수도 있을 것 같아요! 👀

Reply:
- `Map`에서 `get`으로 값을 꺼내왔을 때 `handlerExecution`이 `null` 일 수 있다고 생각됩니다!

- `handlerExecution == null` 대신 `Objects.isNull(handlerExecution)`를 사용한 이유
  - 우선 null 체크가 의도된것임을 명확하게 드러내기 위함. (== null은 그자체로 비교연산이기에)
  - null 체크를 더 읽기 쉽게 하고 명시적인 의미를 부여하기 위한 메서드로 isNull 메서드가 Java 8에서 추가되어서 자주 사용할려고 습관을 들임

Reply:
`Objects.isNull()`의 `API Notes`를 확인해보면 `isNull()` 함수의 등장 배경을 이렇게 설명하고 있어요.

<img width="605" alt="image" src="https://github.com/user-attachments/assets/f92432f4-2fb9-4022-bbef-272ab0eb4dc7">

<br>

`(this::isNull())`과 같은 `Predicate`로 사용되기 위해 만들어진 메서드라는 이 설명 때문에
`null` 체크시 `isNull()`을 쓰는 게 낫다 vs 그래도 `null ==` 연산이 낫다는 갑론을박이 이어지고 있다고 알고 있는데요!

의견이 분분한 주제인 만큼 이것도 뭐가 절대적으로 낫다~ 는 이야기는 불가능하겠네요!

- 의도 표현
- 가독성
- 습관... 

적절한 이유만 있다면 괜찮을 것 같아요 👍

아래는 관련 문서들이니 관심 있으시면 한 번 읽어보셔도 좋을 것 같아용~~~ ☺️

[Oracle 문서](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html)
[스택 오버플로우 - 두 연산을 비교](https://stackoverflow.com/questions/37972859/java-util-objects-isnull-vs-object-null)
[데브위키 - 자바의 null 체크](https://wiki.yowu.dev/ko/dev/Java/if-statement-null-check-variable-null-vs-objects-isnull)

Reply:
함수의 등장 배경과 여러 참고 문서까지...감사합니다 👍👍👍👍
여러 문서를 참고해서 자신만의 기준을 세우는 것이 중요할 것 같네요! 🙇‍♂️